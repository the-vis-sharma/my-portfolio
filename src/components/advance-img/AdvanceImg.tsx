import React, { CSSProperties } from "react";

interface AdvanceImgProps {
  src: string;
  srcWebp: string;
  altText: string;
  style?: CSSProperties | undefined;
}

const AdvanceImg = (props: AdvanceImgProps) => {
  const { src, srcWebp, altText, style } = props;
  return (
    <picture>
      <source srcSet={srcWebp} type="image/webp" />
      <source srcSet={src} type="image/jpeg" />
      <img src={src} alt={altText} style={style ?? {}} />
    </picture>
  );
};

export default AdvanceImg;
