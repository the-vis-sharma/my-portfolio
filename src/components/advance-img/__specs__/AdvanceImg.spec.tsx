import { render } from "@testing-library/react";
import AdvanceImg from "../AdvanceImg";

it("should verify snapshot without style", () => {
  const { asFragment } = render(
    <AdvanceImg src="logo.jpg" srcWebp="logo.webp" altText="test image" />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should verify snapshot with style", () => {
  const { asFragment } = render(
    <AdvanceImg
      src="logo.jpg"
      srcWebp="logo.webp"
      altText="test image"
      style={{ width: "100%" }}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
