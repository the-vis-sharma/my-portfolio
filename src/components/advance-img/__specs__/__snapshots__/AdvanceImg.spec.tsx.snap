// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should verify snapshot with style 1`] = `
<DocumentFragment>
  <picture>
    <source
      srcset="logo.webp"
      type="image/webp"
    />
    <source
      srcset="logo.jpg"
      type="image/jpeg"
    />
    <img
      alt="test image"
      src="logo.jpg"
      style="width: 100%;"
    />
  </picture>
</DocumentFragment>
`;

exports[`should verify snapshot without style 1`] = `
<DocumentFragment>
  <picture>
    <source
      srcset="logo.webp"
      type="image/webp"
    />
    <source
      srcset="logo.jpg"
      type="image/jpeg"
    />
    <img
      alt="test image"
      src="logo.jpg"
    />
  </picture>
</DocumentFragment>
`;
